### ISSUE TITLE
resume from sleep

### ISSUE THREAD

**Issue by janondrusek** [pgj/freebsd-wifibox#2]
Hey!

I have wifibox successfully running on a Dell Vostro 5470 w/ FreeBSD 13. WiFI won't reconnect automatically after resume from sleep. Manually restarting wifibox helps, but it's kinda inconvenient.

Can I get some pointers on how to debug this?

Thanks!

**Comment by pgj** [comment]
Do you have `devd(8)` configured to call `wifibox resume` command on resume?  Note that this is not done automatically on install.

**Comment by JeanVEGA** [comment]
Yes, I also tried manually running `wifibox resume` to no effect:

```
$ cat /usr/local/etc/devd/wifibox.conf
# This is a devd(8) configuration file to run the resume action of
# wifibox on the ACPI resume event.

notify 11 {
        match "system"          "ACPI";
        match "subsystem"       "Resume";
        action "/etc/rc.resume acpi $notify && /usr/local/sbin/wifibox resume";
};

$ service devd status
devd is running as pid 30082.
```

**Comment by pgj** [comment]
So you are saying that `wifibox restart` works for you after resume, but `wifibox resume` does not?  In that case, you can try using the `restart` command in the `devd(8)` configuration file.  My other idea is to insert a call to `logger(1)` at the `action`, something like this:

```
notify 11 {
        match "system"          "ACPI";
        match "subsystem"       "Resume";
        action "/etc/rc.resume acpi $notify && logger "Restarting wifibox" && /usr/local/sbin/wifibox restart";
};
```

Then you can check it in `/var/log/messages` if the action has been called at all.

It may also happen that you are have something attached to the ACPI resume event on the side of `devd(8)` but with a higher priority (that is `11` for wifibox, which can override the default `10` coming from the default configuration).

**Comment by pgj** [comment]
Was I able to help you in resolving the issue?

**Comment by janondrusek** [comment]
Yes, `restart` as a workaround works. It's annoying it takes ~30s to fully restart wifibox. Is there any way to debug this so I can have working `resume`?

Thanks!

**Comment by pgj** [comment]
Sorry for replying so late @janondrusek I did not notice your response.  I have checked what is going on, and it seems `restart` is only needed because the `tap(4)` interface is not brought up after resume.  This can be fixed by setting the `net.link.tap.up_on_open` `sysctl(8)` variable to `1`.  But I am planning to embed this functionality into `wifibox` so one would not need to bother with configuring this.
